packages/xo-web/src/xo-app/sr/tab-advanced.js

```diff
@@ -54,59 +61,80 @@ const UnhealthyVdiChains = flowRight(
-          />
-        )}
-        <TabButton btnStyle='danger' handler={deleteSr} handlerParam={sr} icon='sr-remove' labelId='srRemoveButton' />
-      </Col>
-    </Row>
-    <Row>
-      <Col>
-        <table className='table'>
-          <tbody>
-            <tr>
-              <th>{_('provisioning')}</th>
-              <td>{defined(sr.allocationStrategy, _('unknown'))}</td>
-            </tr>
-            <tr>
-              <th>{_('customFields')}</th>
-              <td>
-                <CustomFields object={sr.id} />
-              </td>
-            </tr>
-          </tbody>
-        </table>
-      </Col>
-    </Row>
-    <Row>
-      <Col>
-        <UnhealthyVdiChains sr={sr} />
-      </Col>
-    </Row>
-  </Container>
-)
+// const SM = props.sm
+
+export default class SrDisks extends Component {
+  render() {
+    const sr = this.props.sr
+    getSmFromSr(sr).then(data => {
+      this.setState({ sm: data })
+    })
+
+    return (
+      <Container>
+        <Row>
+          <Col className='text-xs-right'>
+            <TabButton
+              btnStyle='primary'
+              handler={reclaimSrSpace}
+              handlerParam={sr}
+              icon='sr-reclaim-space'
+              labelId='srReclaimSpace'
+            />
+            {sr.inMaintenanceMode ? (
+              <TabButton
+                btnStyle='warning'
+                handler={toggleSrMaintenanceMode}
+                handlerParam={sr}
+                icon='sr-disable'
+                labelId='disableMaintenanceMode'
+              />
+            ) : (
+              <TabButton
+                btnStyle='warning'
+                handler={toggleSrMaintenanceMode}
+                handlerParam={sr}
+                icon='sr-enable'
+                labelId='enableMaintenanceMode'
+              />
+            )}
+            <TabButton
+              btnStyle='danger'
+              handler={deleteSr}
+              handlerParam={sr}
+              icon='sr-remove'
+              labelId='srRemoveButton'
+            />
+          </Col>
+        </Row>
+        <Row>
+          <Col>
+            <table className='table'>
+              <tbody>
+                <tr>
+                  <th>{_('provisioning')}</th>
+                  <td>{defined(sr.allocationStrategy, _('unknown'))}</td>
+                </tr>
+                <tr>
+                  <th>{_('supportedImageFormats')}</th>
+                  <td>{defined(this.state.sm?.supported_image_formats?.toString(), _('unknown'))}</td>
+                </tr>
+                <tr>
+                  <th>{_('customFields')}</th>
+                  <td>
+                    <CustomFields object={sr.id} />
+                  </td>
+                </tr>
+              </tbody>
+            </table>
+          </Col>
+        </Row>
+        <Row>
+          <Col>
+            <UnhealthyVdiChains sr={sr} />
+          </Col>
+        </Row>
+      </Container>
+    )
+  }
+}
```

Comment:
Use `reaclette` instead.
And do your call inside computed:
```js
 computed: {
   sm: ({sr}) => getSmFromSr(sr)
 }
```
From what I remember, `sm` will be undefined until the promise is resolved.

So you can adapt your line:
`defined(sm?.supported_image_formats?.toString(), _('unknown'))`